client/src/hooks/useLeague.js

```diff
@@ -0,0 +1,14 @@
+import { useQuery } from 'react-query'
+import { getLeague } from 'api'
+import { extractData } from 'utils'
+import { useRouter } from 'next/router'
+
+const useLeagues = (options = {}) => {
+    const router = useRouter()
+    const leagueId = router.query.leagueId
+    const { data, isLoading, error } = useQuery(['league', leagueId], getLeague, options)
+    const league = extractData(data)
+    return { league, isLoading, error }
+}
+
+export default useLeagues
```

Comment:
Even though this makes no difference at all, would be nice if its changed to useLeague instead of useLeagues just cause :))